src/aap_eda/tasks/orchestrator.py

```diff
@@ -121,43 +147,240 @@ def _run_request(
     return True
 
 
-def _make_user_request(
+def dispatch(
     process_parent_type: ProcessParentType,
-    id: int,
-    request_type: ActivationRequest,
-) -> None:
-    """Enqueue a task to manage the activation with the given id."""
-    requests_queue.push(process_parent_type, id, request_type)
-    job_id = _manage_process_job_id(process_parent_type, id)
-    unique_enqueue("activation", job_id, _manage, process_parent_type, id)
+    process_parent_id: int,
+    request_type: Optional[ActivationRequest],
+):
+    job_id = _manage_process_job_id(process_parent_type, process_parent_id)
+    LOGGER.info(
+        f"Dispatching request type {request_type} for {process_parent_type} "
+        "{process_parent_id}",
+    )
+
+    # new processes
+    if request_type in [
+        ActivationRequest.START,
+        ActivationRequest.AUTO_START,
+    ]:
+        LOGGER.info(
+            f"Dispatching {process_parent_type} "
+            "{process_parent_id} as new process.",
+        )
+
+        queue_name = get_most_free_queue_name()
+    else:
+        queue_name = get_queue_name_by_parent_id(
+            process_parent_type,
+            process_parent_id,
+        )
+
+        # If the queue is old or doesn't exist, we use a valid one
+        # to make sure the request is processed and the restart
+        # policy is respected. Same if the request is restart and
+        # the queue is not healthy.
+        if not queue_name or queue_name not in settings.RULEBOOK_WORKER_QUEUES:
+            LOGGER.warning(
+                f"Queue {queue_name} doesn't exist or is not valid. "
+                f"Rescheduling {process_parent_type} {process_parent_id}"
+                f" to the most free queue.",
+            )
```

Comment:
If a deployment changes the queue names we shouldn't be logging a warning every time we encounter an old name.   In fact we shouldn't be logging anything about it during runtime as we'll still process the job.

A single warning at start up to indicate the situation should be sufficient.  We should also issue a warning for a smaller number of queues as that could have a performance impact.

Now...how do we handle that intelligently?  If I start with queues A and B, restart with queues C and D then restart again with C and D that second restart shouldn't generate a warning as nothing's changed.  Similarly for queue count changes.  Maybe we ...

1. clean out all RulebookProcessQueue entries for non-existent queues
2. update the queue and job relation when the job is re-run

The first gets rid of the problem in one shot.
The second cleans it up over time assuming all jobs get re-run, but spits out the warning at startup until then; always if even a single job is never re-run

Reply:
I think the warning message is relevant because that change might be expected or not, so to me is relevant to keep a record. Any other idea or improvement can be discussed in later iterations. 
BTW, there is no relation between a queue and a job, but between a queue and some rulebook_process. Every new process will have a new record with the queue.

Reply:
By 'job' I mean the higher conceptual aspect of every instance run for a particular activation/event_stream.
I have a  mental bias to read "X process" as a singular instance rather than the aggregate.  Anyhow...
for clarity when you say "new process" you mean the aggregate, correct?  Having one per instance would be redundant it would seem.

Reply:
> I think the warning message is relevant because that change might be expected or not, 

How could that occur at runtime?  Yes, the queue could develop a problem and such should be logged but the queue name is not going to disappear from `settings.RULEBOOK_WORKER_QUEUES`; it just won't be there from the beginning.  What's the value in continually logging that?

The lack of a queue name would be odd and would imply some previous error in tracking content.  So that occurrence would be reasonable to log, though mostly for our usage I would think.

Reply:
I don't understand this discussion, honestly. It is not continually logging anything, it is logged just one time. I expect a queue, but the queue is not configured in the system and I consider that it must be logged, because it can be on purpose or not and to me is enough exceptional to be warning. Once the activation is moved to a valid queue all is fine. 

It would be fine if I move it to INFO?

Reply:
I've changed it to INFO level